### ISSUE TITLE
Confusing error in CLI when a blocking invoke fails

### ISSUE THREAD

**Issue by markusthoemmes** [openwhisk/openwhisk#1394]
Blocking activations fail with a very confusing error.

## Environment details:

any environment

## Steps to reproduce the issue:

1. launch an action that fails with `blocking=true`

## Provide the expected results and outputs:

something like:

```
error: Invoked the action 'myNamespace/myAction', but it errored:
{
...
}
```

## Provide the actual results and outputs:

```
stderr: error: Unable to invoke action 'myNamespace/myAction': The connection failed, or timed out. (HTTP status code 502)
```

**Comment by mdeuser** [comment]
Possibly duplicate of #1353 ?

**Comment by markusthoemmes** [comment]
Partly agree, though #1353 seems to be more geared towards the internals? I think this is just a bug in the response handling in the CLI.

**Comment by mdeuser** [comment]
Which invoker runtime were you using when this issue was encountered?

I'm pretty sure the CLI encountered a 502 HTTP response code (similar to #1353).  Would you please run the same command with the `-d` option and include the CLI output.

**Comment by markusthoemmes** [comment]
Yes the CLI encountered a 502. That's what the controller returns for a blocking invocation that fails with an `application developer error`. It's specified like that in swagger.

**Comment by mdeuser** [comment]
In addition to the `-d` CLI dump, would you also include the action code you used when encountering this issue.  If the code is not nodejs based and is not using a rejected Promise to return error results, then this is a separate issue.

**Comment by mdeuser** [comment]
@dubeejw - I was able to recreate this failure with the following action.  The CLI is apparently not handling the error parsing well when the error response is an object.  The HTTP response successfully returns the error object, but the CLI has a parsing bug that results in the error `error: Unable to invoke action 'whiskrejectobj': The connection failed, or timed out. (HTTP status code 502)`

whiskrejrectobj.js
```
function main(message) {
    return Promise.reject({
        name: 'myname',
        aflag: true,
        reason: 'failure cause'
    });
}
```

**Comment by markusthoemmes** [comment]
Thanks for doing my job and reproducing the error @mdeuser :+1:

**Comment by mdeuser** [comment]
Fixed by #1405
